src/main/src/main/java/org/geoserver/catalog/WMSStoreInfo.java

```diff
@@ -28,4 +28,16 @@ public interface WMSStoreInfo extends HTTPStoreInfo {
      * @throws IOException Any I/O problems.
      */
     WebMapServer getWebMapServer(ProgressListener listener) throws IOException;
+
+    String getHeaderName();
```

Comment:
I do not understand why only a single header name / value pair is supported?

Reply:
That is what was previously supported for WMTS, and this PR was supposed to simply align the WMS data store with the WMTS data store (but it morphed into a (single) AuthKey parameter too).

I think it is correct that only a single username/password/authkey/HTTP header is catered for, as that is typically all that is required for authentication.  I might be wrong, but let's then cater for that when there is a firm requirement for it.